libnavigation-core/src/main/java/com/mapbox/navigation/core/history/MapboxHistoryReader.kt

```diff
@@ -0,0 +1,45 @@
+package com.mapbox.navigation.core.history
+
+import com.mapbox.navigation.core.history.model.HistoryEvent
+import com.mapbox.navigation.core.history.model.HistoryEventMapper
+import com.mapbox.navigator.HistoryReader
+
+/**
+ * Allows you to read history files previously saved by [MapboxHistoryRecorder].
+ * All files in the [MapboxHistoryRecorder.fileDirectory] can be read with this reader.
+ *
+ * @param filePath absolute path to a file containing the native history file.
+ */
+class MapboxHistoryReader(
+    val filePath: String
+) : Iterator<HistoryEvent> {
+
+    private val nativeHistoryReader = HistoryReader(filePath)
+    private val historyMapper = HistoryEventMapper()
+
+    private var next: HistoryEvent? = null
+
+    /**
+     * Returns `true` if the iteration has more elements.
+     */
+    override fun hasNext(): Boolean {
+        if (next == null) {
+            val historyRecord = nativeHistoryReader.next()
+            next = if (historyRecord != null) {
+                historyMapper.map(historyRecord)
+            } else {
+                null
+            }
+        }
+        return next != null
+    }
+
+    /**
+     * Returns the next element in the iteration.
+     *
+     * @throws NullPointerException if [hasNext] was not called or did not return true
+     */
+    override fun next(): HistoryEvent = next!!.also {
+        this.next = null
+    }
```

Comment:
This makes a requirement to call `hasNext` before calling `next` which again isn't how `Iterator` interface should be implemented. We'll have to refactor this override as well.

Reply:
you're right, we'll need to load the first event upon creating this. also an easy fix 🤹 

would like to capture these with tests so will also make a wrapper class

Reply:
going to follow up on tests over here https://github.com/mapbox/mapbox-navigation-android/issues/4492